### PR TITLE
feat: V11 migration for gatsby-theme

### PR DESCRIPTION
**Pull request by aledavila**
v11 migration start

### CODE REVIEW

**Review by jnm2377 — APPROVED** [review]

**Review by sstrubberg — CHANGES_REQUESTED** [review]
Holy cow.. this is incredible work. Thank you @aledavila and @jnm2377 for doing this. Two things came to mind as I was reviewing this.
1) The Header needs a little touch up.
header
2) Have you all updated the Example site to complement this upgrades?

File: packages/example/src/pages/guides/styling.mdx [hunk 1]
```diff
@@ -25,12 +25,12 @@ for you to share your component with other theme consumers down the line.
 Every Sass file in your project automatically has access to the the following
```
Comment (joshblack) [inline packages/example/src/pages/guides/styling.mdx:25]: Were you able to "auto-import" styles with Sass modules to mirror how the theme currently works?

File: packages/gatsby-theme-carbon/src/components/ArticleCard/article-card.scss [hunk 2]
```diff
@@ -1,132 +1,141 @@
-.#{$prefix}--article-card .#{$prefix}--tile {
+// $carbon--white-0
```
Comment (joshblack) [inline packages/gatsby-theme-carbon/src/components/ArticleCard/article-card.scss:1]:
```suggestion
```

Small clean-up nit, the comments are great while developing but may be better off excluded from the final code.

File: packages/example/src/pages/components/Aside.mdx [hunk 3]
```diff
@@ -19,9 +19,8 @@ It should only be used within a `<Column>` component with specific parameters.
 **Warning:** If you use more than one `<Aside>` component on a single page, you
 need to provide an accessibility label so that someone using assistive
 technology can quickly understand the purpose of the landmark. See
-[Mozilla Developer Network’s documentation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Complementary_role#Labeling_landmarks)
-for more information. Props for accessibility labels are described in the
-[Props section](#props) below.
+Mozilla Developer Network’s documentation for more information. Props for accessibility labels are described in the
```
[the remark this answers is not in the document]
Comment (aledavila) [reply packages/example/src/pages/components/Aside.mdx:22]: So the v11 inline notifications don't allow for links inside of them anymore for accessibility issues. I'm not sure what we should do in this case.

[the remark this answers is not in the document]
Comment (tay1orjones) [reply packages/example/src/pages/components/Aside.mdx:22]: `ActionableNotification` can be used, it accepts an `inline` prop.

File: packages/example/src/pages/components/InlineNotification.mdx [hunk 4]
```diff
@@ -6,7 +6,7 @@ description: Usage instructions for the Inline notification component
 <PageDescription>
 
 The `<InlineNotification>` component is used to communicate important
-information to a user.
+information to a user. For accessibility reasons, `<InlineNotification>` content cannot include interactive elements, just text.
```
Comment (tay1orjones) [inline packages/example/src/pages/components/InlineNotification.mdx:9]: `ActionableNotification` accepts an `inline` prop that can be used here. It should mostly be a drop-in replacement from a usage perspective.

File: packages/gatsby-theme-carbon/src/styles/index.scss [hunk 5]
```diff
@@ -1,27 +1,29 @@
...
+@use '../components/FeatureCard/feature-card.scss';
+@use '../components/ImageCard/image-card.scss';
 
 //---------------------------------------
 // Included last to ensure that class
 // names take precedence over includes
 // inside component class names
 //---------------------------------------
-@include carbon--type-classes();
+// @include react.type-classes();
```
Comment (tay1orjones) [inline packages/gatsby-theme-carbon/src/styles/index.scss:29]: Do we need something here? The type looks fine on the preview site to me.

File: packages/gatsby-theme-carbon/src/styles/internal/_carbon-components.scss [hunk 6]
```diff
@@ -1,43 +1,43 @@
...
+// @use '@carbon/react/scss/components/progress-indicator/progress-indicator';
+// @use '@carbon/react/scss/components/breadcrumb/breadcrumb';
+// @use '@carbon/react/scss/components/toolbar/toolbar';
+// @use '@carbon/react/scss/components/time-picker/time-picker';
+// @use '@carbon/react/scss/components/slider/slider';
+// @use '@carbon/react/scss/components/tile/tile';
+// @use '@carbon/react/scss/components/skeleton/skeleton';
+// @use '@carbon/react/scss/components/inline-loading/inline-loading';
+// @use '@carbon/react/scss/components/pagination-nav/pagination-nav';
+// @use '@carbon/react/scss/components/ui-shell/ui-shell';
```
Comment (tay1orjones) [inline packages/gatsby-theme-carbon/src/styles/internal/_carbon-components.scss:43]: Did we need to have these in, or are the per-component styles brought in okay through the main entrypoint?

File: packages/gatsby-theme-carbon/src/styles/internal/resources.scss [hunk 7]
```diff
@@ -1,6 +1,17 @@
...
+// $prefix
+@use '@carbon/react/scss/config' as *;
+// $spacing-0X
+@use '@carbon/react/scss/spacing' as *;
+// carbon--type-style
+@use '@carbon/react/scss/type' as *;
+// $standard-easing $transition-x
+@use '@carbon/react/scss/motion' as *;
+// rem() em()
+// @use '@carbon/react/scss/utilities/convert' as *;
```
Comment (tay1orjones) [inline packages/gatsby-theme-carbon/src/styles/internal/resources.scss:17]: This should be available now if we want to continue using it.

**Review by tay1orjones — COMMENTED** [review]
This looks great! The deploy preview doesn't seem to have any problems. Below is just a few items I'm curious about.

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**